controller/service.rst

```diff
@@ -80,6 +80,98 @@ If your controller implements the ``__invoke()`` method - popular with the
 Action-Domain-Response (ADR) pattern, you can simply refer to the service id
 (``AppBundle\Controller\HelloController`` or ``app.hello_controller`` for example).
 
+If you want to use the ADR pattern rather than the default controller approach, you can use
+the new features provided by the container like public/private injections and autowiring, 
+in order to work, you must update the services.yml file ::
+
+    # ...
+
+    services:
+        _defaults:
+            autowire: true
+            autoconfigure: true
+            public: false
+            
+        # Allow to load every actions
+        AppBundle\Action\:
+            resource: '../../src/AppBundle/Action/'
+            public: true
+
+Once the file is updated, delete your Controller folder and create an Action one then 
+build a HelloAction class that use the ADR principes ::
+
+    <?php
+
+    namespace AppBundle\Action;
+
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Templating\EngineInterface;
```

Comment:
This component is to be deprecated, a Twig `Environment` should be used instead.